### DISCUSSION TITLE
[firebase_performance] Missing web support

### DISCUSSION THREAD

**Discussion by creativecreatorormaybenot** [firebase/flutterfire#6649] [Feature request] [answered]
Performance Monitoring for web is in beta, see https://firebase.google.com/docs/perf-mon/get-started-web.

It already supports monitoring:

* [Page load](https://firebase.google.com/docs/perf-mon/page-load-traces) (automatically)
* [HTTP requests](https://firebase.google.com/docs/perf-mon/network-traces?platform=web) (automatically)
* [Custom traces](https://firebase.google.com/docs/perf-mon/custom-code-traces)

This means most functionality is supported and more importantly, the supported functionality is extremely useful.

Currently, we have to write custom logic:

```dart
if (kIsWeb) {
  // Manually call Firebase Performance JavaScript SDK.
} else {
  // Use FlutterFire plugin that only works on mobile.
}
```

It would be extremely useful if the custom traces would just work on web as well.

**Comment by creativecreatorormaybenot** [comment]
@Salakar We want to use this feature and we will write a custom wrapper to either call web or mobile. 

I thought about contributing the web implementation (that is what I would want to do), but [I just checked](https://github.com/FirebaseExtended/flutterfire/tree/ee3913a86ae4ca4ea01ad884cda11e0a3369bc97/packages/firebase_performance) and there is no `platform_interface`, which makes it unfeasible 😕

**Reply by creativecreatorormaybenot** [reply]
cc @russellwheatley

**Comment by yalunqin** [comment, marked as answer]
@creativecreatorormaybenot we have a [PR](https://github.com/FirebaseExtended/flutterfire/pull/6920) adding the web support and it is under review (@russellwheatley).

**Reply by Salakar** [reply]
PR has now been merged
